provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -28,6 +28,46 @@ import (
 	"github.com/vmware/govmomi/vim25/types"
 )
 
+type DiskProvisioningType string
+
+const (
+	// DiskTypeThin sets the provisioning type for disks, when cloning
+	// from a template, to "thin". This is also known as "sparse"
+	// provisioning. Only written blocks inside the virtual disk are
+	// deducted from the underlying datastore.
+	DiskTypeThin DiskProvisioningType = "thin"
+	// DiskTypeThickLazyZero sets the provisioning type for disks to
+	// "Thick Provision Lazy Zeroed". The entire size of the virtual disk,
+	// written and unwritten space, is deducted from the underlying datastore.
+	// Unwritten blocks inside the virtual disk, are not zeroed out. This
+	// speeds up disk cloning but introduces two pitfals:
+	//   * If there is previously written data on the datastore where the
+	//     space is allocated, it can be recovered in the newly instantiated
+	//     machine that now ocupies that space. This can be a potential
+	//     security risk.
+	//   * Before new data will be written to a disk area, the hypervisor
+	//     will first zero out the disk area before writing to it. This adds
+	//     latency to disk writes in that area.
+	DiskTypeThickLazyZero DiskProvisioningType = "thickLazyZero"
```

Comment:
Let's go with lower case words separated by a dash instead of camel case for the values.

Reply:
Per the comment above, please add that DiskTypeThick is the default.

Reply:
I defined a variable indicating the default, here: https://github.com/juju/juju/pull/12794/files#diff-31c3de9d4ede680560b714357df5e54ba8652ab428941b320e5dc5a81005a586R68

But I will also add a note in the comment here.

Reply:
Done